File: src/main/java/com/dm/data/repository/GroupRepository.java
package com.dm.data.repository;

import com.dm.data.entity.GroupEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

/**
 * Repository for GroupEntity.
 */
@Repository
public interface GroupRepository extends JpaRepository<GroupEntity, Long> {

    Optional<GroupEntity> findByCode(String code);

    java.util.List<GroupEntity> findBySpecializationId(Long specializationId);

    java.util.List<GroupEntity> findByStudyYear(Integer studyYear);

    java.util.List<GroupEntity> findByIsModular(Integer isModular);

    java.util.List<GroupEntity> findBySpecialization_Department_Id(Long departmentId);
}

File: src/main/java/com/dm/data/repository/GroupRepository.kt
package com.dm.data.repository

import com.dm.model.Group
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface GroupRepository : JpaRepository<Group, Long> {

    fun findByName(name: String): Group?

    fun findAllByStudyYear(studyYear: Int): List<Group>

    fun findAllBySpecialization(specialization: String): List<Group>

    fun findAllByFaculty(faculty: String): List<Group>
}